ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlMapDecoder.kt

```diff
@@ -99,4 +126,21 @@ public class TomlMapDecoder(
             kotlinxIndex += 2
         }
     }
+
+    public companion object {
+        /**
+         * @param deserializer - deserializer provided by Kotlin compiler
+         * @param rootNode - root node for decoding (created after parsing)
+         * @param config - decoding configuration for parsing and serialization
+         * @return decoded (deserialized) object of type T
+         */
+        public fun <T> decode(
```

Comment:
So what was the problem? You have introduced a new decoder?

Reply:
Example from test `testNestedRootMapDecoder`, we have this toml:
```
my_key_1 = "my_value_1"
my_key_2 = "my_value_2"
```

And we got Exception `my_key_1 is not a valid map index` in TomlMainDecoder::decodeElementIndex. **rootNode** has 2 TomlKeyValuePrimitive but no TomlTable (ofc, we can't know it on parsing level and anyway we don't have TomlTable.name)

In that case we can:
- Use `TomlMapDecoder.decode()` instead of `TomlMainDecoder.decode()` in `Toml` class after parsing if we have descriptor.kind == StructureKind.MAP
- Do the same in `TomlMainDecoder`
- ? smth else

I decided to make the first option just because it's easier :)

Reply:
Yeah, sounds very reasonable, LGTM!

Reply:
I don't have strong opinion so far, how do you think we can avoid TomlMainDecoder?